pkg/api/v1/atlasnetworkpeering_types.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2020 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/provider"
+
+// +k8s:deepcopy-gen=false
+
+type AtlasNetworkPeeringConfig struct {
+	//ProviderName is the name of the provider. If not set, it will be set to "AWS".
+	// +optional
+	ProviderName provider.ProviderName `json:"providerName,omitempty"`
```

Comment:
I'm wondering about the value of using this type. I decided to not use it in private endpoint CRD

Reply:
Can you elaborate?
I mean, this and the other config will go embedded in the spec and are the part of the API that directly map to Atlas concepts and NOT Kubernetes related fields.
What did you drop on private endpoints exactly?

Reply:
The `provider.ProviderName` as the type for provider value(AWS/AZURE/GCP)

Reply:
ok, so you just mean make the provided name here a plain string?

Reply:
Most importantly. Is `ProviderName` already used in the existing subobject structs?

Reply:
I can revisit usages, but if we must be consistent we should be across also private endpoints right?